sql/sqlserver/server.go

```diff
@@ -304,57 +306,46 @@ func (s *Server) Use(session *Session, p *parser.Use, args []sqlwire.Datum, resp
 }
 
 // CreateDatabase creates a database if it doesn't exist.
-func (s *Server) CreateDatabase(session *Session, p *parser.CreateDatabase, args []sqlwire.Datum, resp *sqlwire.Response) {
+func (s *Server) CreateDatabase(session *Session, p *parser.CreateDatabase, args []sqlwire.Datum, resp *sqlwire.Response) error {
 	if p.Name == "" {
-		resp.SetGoError(errors.New("empty database name"))
-		return
+		return errEmptyDatabaseName
 	}
 
 	nameKey := keys.MakeNameMetadataKey(structured.RootNamespaceID, strings.ToLower(p.Name))
 	if gr, err := s.db.Get(nameKey); err != nil {
-		resp.SetGoError(err)
-		return
+		return err
 	} else if gr.Exists() {
 		if p.IfNotExists {
-			return
+			return nil
 		}
-		resp.SetGoError(fmt.Errorf("database \"%s\" already exists", p.Name))
-		return
+		return fmt.Errorf("database \"%s\" already exists", p.Name)
```

Comment:
i don't think it's so important here. this error will wind up going back to the client, and the client need not care about the line number.